joshwcomeau/guppy
# Fix taskkill issue 364 & add tests

**Related Issue:**
#364 & #344

**Summary:**
- Changed to `spawnSync`  so taskkill is executed before closing the app.
- Removed `event.preventDefault` from ipc callback as I think it's not available & not needed. IpcMain event got just two key - see the docs [here](https://electronjs.org/docs/api/ipc-main#event-object)

Killing tested on Ubuntu  18.0.4 (64bit) & Windows 10 Pro (64bit).

**Test steps to verify task killing**
1. Launch app with `yarn start` (executing the binary should do the same)
2. Launch a devServer (will create a listening port on 3001)
3. Verify that server is running with browser or console (Windows `netstat -aon | find "3001"` or `lsof -i :3001` on Mac/Linux)
4. Close app with x-button and close on app icon
5. Repeat step 3 (verify that the server is stopped)

**Discussion**
- Does changing to `spawnSync` has an impact on performance? How can we check this? Maybe starting multiple devServers and check app close duration?
- How can we test ipcMain & ipcRenderer? I'd like to add a test but I haven't got it working yet. I'd like to check that adding / deleting of ids is working and that taskkill is called. OK, testing is working but it is probably not a perfect solution. Anyway, I think it's OK.
